src/bin/bobd.rs

```diff
@@ -70,45 +74,106 @@ impl server::BobApi for BobSrv {
+        let sw = Stopwatch::start_new();
+        let param = req.into_inner();
+        if !Self::get_is_valid(&param) {
+            warn!("GET[-] invalid arguments - key and data is mandatory");
+            Box::new(future::err(tower_grpc::Status::new(
+                tower_grpc::Code::InvalidArgument,
+                "Key and data is mandatory",
+            )))
+        } else {
+            let key = BobKey {
+                key: param.key.unwrap().key,
+            };
+
+            Box::new(
+                self.grinder
+                    .get(key, {
+                        let mut opts: BobOptions = Default::default();
+                        if let Some(vopts) = param.options.as_ref() {
+                            if vopts.force_node {
+                                opts |= BobOptions::FORCE_NODE;
+                            }
+                        }
+                        opts
+                    })
                     .then(move |r| {
                         let elapsed = sw.elapsed_ms();
                         match r {
                             Ok(r_ok) => {
                                 info!(
-                                    "PUT[{}]-OK local:{} ok dt: {}ms",
-                                    key, !r_ok.is_clustered, elapsed
+                                    "GET[{}]-OK local:{} dt: {}ms",
+                                    key,
+                                    r_ok.is_local(),
+                                    elapsed
                                 );
-                                ok(Response::new(OpStatus { error: None }))
+                                ok(Response::new(Blob {
+                                    data: match r_ok {
+                                        ServeTypeOk::Cluster(r) => r.result.data,
+                                        ServeTypeOk::Local(r) => r.data,
+                                    },
+                                }))
                             }
                             Err(r_err) => {
                                 error!(
-                                    "PUT[{}]-ERR local:{}  dt: {}ms {}",
+                                    "GET[{}]-ERR local:{}  dt: {}ms {:?}",
                                     key,
-                                    !r_err.is_clustered,
+                                    r_err.is_local(),
                                     elapsed,
-                                    r_err.to_string()
+                                    r_err
                                 );
-                                ok(Response::new(OpStatus {
-                                    error: Some(BobError {
-                                        code: 1,
-                                        desc: format!("Failed to write {}", r_err.to_string()),
-                                    }),
-                                }))
+                                let err = match r_err {
```

Comment:
Looks like you just convert errors, you can define your own `enum Error` and use combinators instead of `match`